### PR TITLE
convert to python3

### PR DESCRIPTION
**Pull request by fawaf**
- update with new irclib
- run 2to3 conversion on python files

### CODE REVIEW

**Comment by gnowxilef** [comment]
fixes https://github.com/jimrollenhagen/pywhatauto/issues/43

File: globals.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #! /usr/bin/env python
 
 from datetime import datetime
-import threading, sys, os, Queue#, thread
+import threading, sys, os, queue#, thread
```
Comment (jimrollenhagen) [inline globals.py:2]: This won't work on python 2, we should support both. We could use six (https://pythonhosted.org/six/) to do this, however that would be a new external dependency, which we don't tend to place on users (though I'm open to changing things up to do this).

Alternatively, could try/except ImportError to do the right thing.

File: irclib.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Copyright (C) 1999--2002  Joel Rosdahl
+# Copyright (C) 1999--2002 Joel Rosdahl
```
Comment (jimrollenhagen) [inline irclib.py:1]: No need to change whitespace, just makes the diff harder to review, and breaks the blame.

File: irclib.py [hunk 3]
```diff
@@ -98,56 +97,56 @@ class IRCError(Exception):
...
+irc = irclib.IRC()
+server = irc.server()
+server.connect(\"irc.some.where\", 6667, \"my_nickname\")
+server.privmsg(\"a_nickname\", \"Hi there!\")
+irc.process_forever()
+
+This will connect to the IRC server irc.some.where on port 6667
+using the nickname my_nickname and send the message \"Hi there!\"
+to the nickname a_nickname.
+"""
```
Comment (jimrollenhagen) [inline irclib.py:123]: Ditto re: whitespace, and all through this file. Python style convention is the way this was before.

File: irclib.py [hunk 4]
```diff
@@ -200,19 +199,19 @@ def process_timeout(self):
...
-        at the process_forever method.
-        """
-        sockets = map(lambda x: x._get_socket(), self.connections)
-        sockets = filter(lambda x: x != None, sockets)
+This method should be called periodically to check and process
+incoming data, if there are any. If that seems boring, look
+at the process_forever method.
+"""
+        sockets = [x._get_socket() for x in self.connections]
+        sockets = [x for x in sockets if x != None]
```
Comment (jimrollenhagen) [inline irclib.py:212]: Why change this?

File: irclib.py [hunk 5]
```diff
@@ -200,19 +199,19 @@ def process_timeout(self):
...
         if sockets:
-            i = select.select(sockets, [], [], timeout)[0]
+            (i, o, e) = select.select(sockets, [], [], timeout)
```
Comment (jimrollenhagen) [inline irclib.py:214]: Ditto, no need to change this

File: irclib.py [hunk 6]
```diff
@@ -363,53 +363,54 @@ class ServerNotConnectedError(ServerConnectionError):
...
+ircname -- The IRC name ("realname").
 
-            localaddress -- Bind the connection to a specific local IP address.
+localaddress -- Bind the connection to a specific local IP address.
 
-            localport -- Bind the connection to a specific local port.
+localport -- Bind the connection to a specific local port.
 
-            ssl -- Enable support for ssl.
+sslsock -- Enable support for ssl.
```
Comment (jimrollenhagen) [inline irclib.py:405]: Now I'm wondering, did you just pull a new version of irclib.py? If so, that's fine, but please note that in the commit message and/or pull request description.

File: torrentparser.py [hunk 7]
```diff
@@ -1,13 +1,13 @@
...
-from __future__ import division
+
+
 
 import re, os, hashlib
 
 def main():
     dir = '/home/blubba/.config/deluge/state'
-    print('%-50s %-3s %10.2s  %-5s' %('Infohash','M','MBs','Name'))
+    print(('%-50s %-3s %10.2s  %-5s' %('Infohash','M','MBs','Name')))
```
Comment (jimrollenhagen) [inline torrentparser.py:10]: No need for double parens here (and other places in the diff)

File: torrentparser.py [hunk 8]
```diff
@@ -1,13 +1,13 @@
 #!/usr/bin/env python
 # -*- coding: UTF-8 -*-
-from __future__ import with_statement
-from __future__ import division
```
Comment (jimrollenhagen) [inline torrentparser.py:4]: Might still need this on 2.x?

File: torrentparser.py [hunk 9]
```diff
@@ -139,7 +139,7 @@ def encode(self, dictionary=None):
         if not dictionary:
             dictionary = self.dictionary
         ben = 'd'
-        for key in sorted(dictionary.iterkeys(),key=str.lower):
+        for key in sorted(iter(dictionary.keys()),key=str.lower):
```
Comment (jimrollenhagen) [inline torrentparser.py:142]: Should just be able to do `dictionary.keys()` here, iirc

File: WHATauto.py [hunk 10]
```diff
@@ -16,16 +16,16 @@
...
+print(('You are running pyWHATauto version %s\n'%VERSION))
 
 #from time import strftime, strptime
 from datetime import datetime, timedelta
 
 from threading import Thread
-from BaseHTTPServer import BaseHTTPRequestHandler, HTTPServer
-from SocketServer import ThreadingMixIn
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from socketserver import ThreadingMixIn
```
Comment (jimrollenhagen) [inline WHATauto.py:29]: Same problems here as below with imports

File: WHATauto.py [hunk 11]
```diff
@@ -16,16 +16,16 @@
...
 
-import db, time, os, re, ConfigParser, thread, urllib, urllib2, random, cookielib, socket, math, traceback, sqlite3, ssl, threading#, WHATparse as WP, #htmllib, 
+import db, time, os, re, configparser, _thread, urllib.request, urllib.parse, urllib.error, urllib.request, urllib.error, urllib.parse, random, http.cookiejar, socket, math, traceback, sqlite3, ssl, threading#, WHATparse as WP, #htmllib,
```
Comment (jimrollenhagen) [inline WHATauto.py:31]: Same import issue

File: WHATauto.py [hunk 12]
```diff
@@ -192,14 +192,14 @@ def loadConfigs():
     
     G.ALIASLENGTH = 0
     longest = ''
-    for val in G.TOALIAS.itervalues():
+    for val in list(G.TOALIAS.values()):
```
Comment (jimrollenhagen) [inline WHATauto.py:199]: no need for `list()`

**Review by jimrollenhagen — CHANGES_REQUESTED** [review]
Thanks for doing this, it's much appreciated :)

There's lots of things to clean up here, noted some of them inline. This also doesn't work on python 2, which I think we should still support.

File: irclib.py (hunk 2 above)
Comment (fawaf) [reply irclib.py:1]: huh, i wonder how that got changed. i think `2to3` probably changed it.

File: irclib.py (hunk 4 above)
Comment (fawaf) [reply irclib.py:212]: `2to3` automagically changed it if i am not mistaken.

File: irclib.py (hunk 5 above)
Comment (fawaf) [reply irclib.py:214]: `2to3` automagically changed it if i am not mistaken.

File: irclib.py (hunk 6 above)
Comment (fawaf) [reply irclib.py:405]: yeah i did. if i didn't, the program failed to load correctly.

File: torrentparser.py (hunk 7 above)
Comment (fawaf) [reply torrentparser.py:10]: good catch.

File: torrentparser.py (hunk 9 above)
Comment (fawaf) [reply torrentparser.py:142]: `2to3` automagically changed it if i am not mistaken. i'll change it back.

File: WHATauto.py (hunk 12 above)
Comment (fawaf) [reply WHATauto.py:199]: `2to3` automagically changed it if i am not mistaken. i'll change it back.

**Pull request closed without merging**